From e8e01e0ef200ca45cbb170cc192ff36fad66ff1b Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Tue, 17 Jun 2014 17:00:10 +0800
Subject: [PATCH 56/95] ENGR00318735 - [sabresd_6dq]: can't take picture on
 kk4.4.3_3.10.y daily build

It's kk4.4.3's issue.
In JpegProcessor::processNewCapture, captureBuffer is defined two times.
One in { }, another in whole function.
But captureBuffer set value in {}, but used outside {}.

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 services/camera/libcameraservice/api1/client2/JpegProcessor.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp b/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp
index 825c41d..e127aed 100644
--- a/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp
+++ b/services/camera/libcameraservice/api1/client2/JpegProcessor.cpp
@@ -14,6 +14,10 @@
  * limitations under the License.
  */
 
+/*
+ * Copyright (C) 2013-2014 Freescale Semiconductor, Inc.
+*/
+
 #define LOG_TAG "Camera2-JpegProcessor"
 #define ATRACE_TAG ATRACE_TAG_CAMERA
 //#define LOG_NDEBUG 0
@@ -272,7 +276,7 @@ status_t JpegProcessor::processNewCapture() {
         }
 
         // TODO: Optimize this to avoid memcopy
-    sp<MemoryBase> captureBuffer = new MemoryBase(mCaptureHeap, 0, jpegSize);
+        captureBuffer = new MemoryBase(mCaptureHeap, 0, jpegSize);
         void* captureMemory = mCaptureHeap->getBase();
         memcpy(captureMemory, imgBuffer.data, jpegSize);
 
-- 
1.8.0

